### PR TITLE
Refactor to use `oblast` lib

### LINKED ISSUE #3: Convert uses of `point::Point` to `oblast::Fr`
@michaelsproul made an awesome wrapper around `blst` primitives, the rest of the library should use them

### PR DESCRIPTION
**Pull request by ralexstokes**
Fixes #3.

Moves `unsafe` code into `oblast` wrapper and refactors usage of `blst` to go through `oblast`.

### CODE REVIEW

File: src/constants.rs [hunk 1]
```diff
@@ -1,7 +1,5 @@
 use num_bigint::BigUint;
 
-pub const MODULUS_BIT_SIZE: usize = oblast::MODULUS_BIT_SIZE;
-
 pub fn get_modulus() -> BigUint {
```
Comment (ralexstokes) [inline src/constants.rs]: at this point, feels like it should be in `oblast`

File: src/commitment.rs [hunk 2]
```diff
@@ -283,20 +266,23 @@ mod tests {
...
+                );
             }
             let expected_commitment_serialization = hex::decode(expected_commitment_hex).unwrap();
             assert_eq!(commitment_serialization, expected_commitment_serialization);
 
             // does proof match?
             let mut proof_serialization = vec![0u8; 48];
             unsafe {
-                blst::blst_p1_compress(proof_serialization.as_mut_ptr(), &opening.proof);
+                blst::blst_p1_compress(proof_serialization.as_mut_ptr(), &opening.proof.point);
```
Comment (ralexstokes) [inline src/commitment.rs]: feeling like adding a serialization method to `oblast::PX` and then making `point` private again...

File: oblast/src/lib.rs [hunk 3]
```diff
@@ -86,7 +178,7 @@ macro_rules! define_curve_struct {
             #[doc = "Point on the curve sub-group " $group_name "."]
             #[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
             pub struct $struct_name {
-                point: blst::[<blst_ $blst_name>],
+                pub point: blst::[<blst_ $blst_name>],
```
Comment (ralexstokes) [inline oblast/src/lib.rs]: i think this is temporary, see: https://github.com/ralexstokes/kzg/pull/4/files#r585164136

File: oblast/src/lib.rs [hunk 4]
```diff
@@ -70,6 +148,20 @@ pub struct Scalar {
...
 
+impl Scalar {
+    /// Construct a `blst_scalar` instance of a `Fr` value from bytes in big-endian order.
+    /// Panics if the value is not in `Fr`.
+    pub fn from_fr_bytes(value: &[u8]) -> Self {
+        assert!(value.len() == 32);
+        let mut scalar = blst::blst_scalar::default();
+        unsafe {
+            blst::blst_scalar_from_bendian(&mut scalar, value.as_ptr());
+            assert!(blst::blst_scalar_fr_check(&scalar));
```
Comment (michaelsproul) [inline oblast/src/lib.rs:161]: Long-term we may want to make this a `debug_assert` for performance reasons, but I'm happy to leave it in for now

File: src/constants.rs (hunk 1 above)
Comment (michaelsproul) [reply src/constants.rs:3]: yeah, agreed

File: src/commitment.rs [hunk 5]
```diff
@@ -1,20 +1,16 @@
-use crate::constants;
-use crate::point;
 use crate::polynomial;
 use crate::setup;
-use blst;
-// TODO: ripple oblast through rest of codebase
-use oblast::{Fr, P1, P2};
+use oblast;
```
Comment (michaelsproul) [inline src/commitment.rs]: I notice you're quite fond of the module-inclusive identifiers like `oblast::Fr`, I prefer it without the module personally but I'm happy to go modular to match your style :grin:

**Review by michaelsproul — COMMENTED** [review]
Nice! Looks good modulo those small tweaks mentioned above

File: src/commitment.rs (hunk 2 above)
Comment (michaelsproul) [reply src/commitment.rs]: good idea!

File: src/commitment.rs (hunk 5 above)
Comment (ralexstokes) [reply src/commitment.rs]: i can do either, in this case it is clear from context so no problems dropping it

File: src/commitment.rs (hunk 2 above)
Comment (ralexstokes) [reply src/commitment.rs]: you had actually already exposed the `compress` functions from `blst` so this was easy!

**Pull request merged**